apps/api/src/app/bridge/usecases/preview-step/preview-step.usecase.ts

```diff
@@ -15,87 +14,110 @@ export class PreviewStep {
 
   async execute(command: PreviewStepCommand) {
     const environment = await this.environmentRepository.findOne({ _id: command.environmentId });
+    const bridgeUrl = this.assertBridgeUrl(command, environment);
+    try {
+      const response = await this.restCallBridgeWithPreviewCommand(command, environment, bridgeUrl);
+      this.assertHasPreviewOutput(response);
+
+      return response.data;
+    } catch (e) {
+      throw this.buildBridgeErrors(e, bridgeUrl);
+    }
+  }
+
+  private assertHasPreviewOutput(
+    response: AxiosResponse
+  ): asserts response is AxiosResponse & { data: { outputs: any; metadata: any } } {
+    if (!response.data?.outputs || !response.data?.metadata) {
+      throw new BadRequestException({
+        code: BridgeErrorCodeEnum.BRIDGE_UNEXPECTED_RESPONSE,
+        message: JSON.stringify(response.data),
+      });
+    }
+  }
+  private async restCallBridgeWithPreviewCommand(
```

Comment:
As discussed, we can replace this Bridge request builder and executor with the [`ExecuteBridgeRequest` use-case](https://github.com/novuhq/novu/blob/next/libs/application-generic/src/usecases/execute-bridge-request/execute-bridge-request.usecase.ts).